### DISCUSSION TITLE
Get Timeline lint error when changing folder name of 'examples/browser'

### DISCUSSION THREAD

**Discussion by chris-odell-focs** [eclipse-theia/theia#9588] [Q&A]
So I want to re-purpose the 'Examples/Browser' part of the build and call it something else. I have renamed it 'bindery/src/' and believe I have made the requisite changes in the package.json file. I checked the other configs but these all seem to be generic.

However when I run yarn, everything builds as expected except I get an error when it comes to building timeline.

@theia/timeline:   0:0  error  Parsing error: "parserOptions.project" has been set for @typescript-eslint/parser.
@theia/timeline: The file does not match your project config: packages/timeline/src/browser/timeline-frontend-module.ts.
@theia/timeline: The file must be included in at least one of the projects provided
@theia/timeline: ✖ 1 problem (1 error, 0 warnings)
@theia/timeline: Exit with failure status (1): if-env SKIP_LINT=true && echo 'skip lint check' || eslint --cache=true --no-error-on-unmatched-pattern=true "{src,test}/**/*.{ts,tsx}"
@theia/timeline: info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
@theia/timeline: error Command failed with exit code 1.
lerna ERR! execute callback with error
lerna ERR! Error: Command failed: yarn run lint
lerna ERR! Exit with failure status (1): if-env SKIP_LINT=true && echo 'skip lint check' || eslint --cache=true --no-error-on-unmatched-pattern=true "{src,test}/**/*.{ts,tsx}"
lerna ERR! error Command failed with exit code 1.
lerna ERR! 
lerna ERR! $ theiaext lint
lerna ERR! $ if-env SKIP_LINT=true && echo 'skip lint check' || eslint --cache=true --no-error-on-unmatched-pattern=true "{src,test}/**/*.{ts,tsx}"
lerna ERR! 
lerna ERR! /home/bobs/workspace/bindery/packages/timeline/src/browser/timeline-frontend-module.ts
lerna ERR!   0:0  error  Parsing error: "parserOptions.project" has been set for @typescript-eslint/parser.
lerna ERR! The file does not match your project config: packages/timeline/src/browser/timeline-frontend-module.ts.
lerna ERR! The file must be included in at least one of the projects provided
lerna ERR! 
lerna ERR! ✖ 1 problem (1 error, 0 warnings)
lerna ERR! 
lerna ERR! info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
lerna ERR! 
lerna ERR!     at /home/bobs/workspace/bindery/node_modules/execa/index.js:236:11
lerna WARN complete Waiting for 1 child process to exit. CTRL-C to exit immediately.
Error: Command failed: yarn run lint
Exit with failure status (1): if-env SKIP_LINT=true && echo 'skip lint check' || eslint --cache=true --no-error-on-unmatched-pattern=true "{src,test}/**/*.{ts,tsx}"
error Command failed with exit code 1.

Obviously I can see that the file cannot be found, but I cannot see for the life of me why it's not being picked up.

I thought it was because I had changed the example/browser folder name but other modules which have a browser folder are being picked up ok.

I have changed the package.json to this : 

{
  "private": true,
  "name": "@theia/monorepo",
  "version": "0.0.0",
  "engines": {
    "yarn": "1.0.x || >=1.2.1",
    "node": ">=12.14.1 <13"
  },
  "resolutions": {
    "**/@types/node": "12",
    "**/node-abi": "^2.18.0"
  },
  "devDependencies": {
    "@types/chai-string": "^1.4.0",
    "@types/jsdom": "^11.0.4",
    "@types/node": "12",
    "@types/sinon": "^2.3.5",
    "@types/temp": "^0.8.29",
    "@types/uuid": "^7.0.3",
    "@typescript-eslint/eslint-plugin": "^4.8.1",
    "@typescript-eslint/eslint-plugin-tslint": "^4.8.1",
    "@typescript-eslint/parser": "^4.8.1",
    "chai-string": "^1.4.0",
    "colors": "^1.4.0",
    "concurrently": "^3.5.0",
    "electron-mocha": "^8.2.0",
    "eslint": "^7.14.0",
    "eslint-plugin-deprecation": "^1.1.0",
    "eslint-plugin-import": "^2.22.1",
    "eslint-plugin-no-null": "^1.0.2",
    "eslint-plugin-no-unsanitized": "^3.1.4",
    "eslint-plugin-react": "^7.21.5",
    "if-env": "^1.0.4",
    "ignore-styles": "^5.0.1",
    "jsdom": "^11.5.1",
    "lerna": "^2.2.0",
    "nyc": "^15.0.0",
    "rimraf": "^2.6.1",
    "sinon": "^3.3.0",
    "temp": "^0.8.3",
    "tslint": "^5.12.0",
    "typedoc": "^0.17.7",
    "typedoc-plugin-external-module-map": "^1.2.1",
    "typescript": "^3.9.2",
    "uuid": "^8.0.0"
  },
  "scripts": {
    "preinstall": "node-gyp install",
    "postinstall": "node scripts/post-install.js",
    "prepare": "yarn prepare:references && yarn --cwd packages/core prepare && yarn prepare:build && yarn prepare:hoisting && yarn download:plugins",
    "prepare:references": "node scripts/compile-references.js",
    "prepare:build": "yarn build && run lint && run build \"@theia/bindery-*\" --stream --parallel",
    "prepare:hoisting": "theia check:hoisted -s",
    "clean": "yarn lint:clean && node scripts/run-reverse-topo.js yarn clean",
    "build": "tsc -b configs/root-compilation.tsconfig.json",
    "watch": "tsc -b configs/root-compilation.tsconfig.json -w",
    "lint": "run lint",
    "lint:clean": "rimraf .eslintcache",
    "lint:oneshot": "node --max-old-space-size=4096 node_modules/eslint/bin/eslint.js --cache=true \"{dev-packages,packages,bindery}/**/*.{ts,tsx}\"",
    "docs": "rimraf gh-pages/docs/next && typedoc --tsconfig configs/typedoc-tsconfig.json --options configs/typedoc.json",
    "test": "yarn test:references && yarn test:theia && yarn test:src",
    "test:references": "node scripts/compile-references --dry-run",
    "test:theia": "run test \"@theia/!(bindery-)*\" --stream --concurrency=1",
    "test:bindery": "yarn rebuild:src && run test \"@theia/bindery\"",
    "rebuild:clean": "rimraf .app_modules",
    "rebuild:bindery": "theia rebuild:bindery",
    "publish": "yarn && yarn test && yarn publish:latest",
    "publish:latest": "lerna publish --exact && yarn publish:check",
    "publish:next": "yarn next:publish && yarn next:publish --skip-npm && yarn publish:check",
    "next:publish": "lerna publish --exact --canary=next --npm-tag=next --yes",
    "publish:check": "node scripts/check-publish.js",
    "start:app": "yarn rebuild:app && yarn --cwd bindery/src start",
    "download:plugins": "theia download:plugins"
  },
  "workspaces": [
    "dev-packages/*",
    "packages/*",
    "bindery/*"
  ],
  "theiaPluginsDir": "plugins",
  "theiaPlugins": {
    "vscode-builtin-bat": "https://open-vsx.org/api/vscode/bat/1.45.1/file/vscode.bat-1.45.1.vsix",
    "vscode-builtin-clojure": "https://open-vsx.org/api/vscode/clojure/1.45.1/file/vscode.clojure-1.45.1.vsix",
    "vscode-builtin-coffeescript": "https://open-vsx.org/api/vscode/coffeescript/1.45.1/file/vscode.coffeescript-1.45.1.vsix",
    "vscode-builtin-configuration-editing": "https://open-vsx.org/api/vscode/configuration-editing/1.45.1/file/vscode.configuration-editing-1.45.1.vsix",
    "vscode-builtin-cpp": "https://open-vsx.org/api/vscode/cpp/1.45.1/file/vscode.cpp-1.45.1.vsix",
    "vscode-builtin-csharp": "https://open-vsx.org/api/vscode/csharp/1.45.1/file/vscode.csharp-1.45.1.vsix",
    "vscode-builtin-css": "https://open-vsx.org/api/vscode/css/1.45.1/file/vscode.css-1.45.1.vsix",
    "vscode-builtin-css-language-features": "https://open-vsx.org/api/vscode/css-language-features/1.45.1/file/vscode.css-language-features-1.45.1.vsix",
    "vscode-builtin-debug-auto-launch": "https://open-vsx.org/api/vscode/debug-auto-launch/1.45.1/file/vscode.debug-auto-launch-1.45.1.vsix",
    "vscode-builtin-docker": "https://open-vsx.org/api/vscode/docker/1.45.1/file/vscode.docker-1.45.1.vsix",
    "vscode-builtin-emmet": "https://open-vsx.org/api/vscode/emmet/1.45.1/file/vscode.emmet-1.45.1.vsix",
    "vscode-builtin-fsharp": "https://open-vsx.org/api/vscode/fsharp/1.45.1/file/vscode.fsharp-1.45.1.vsix",
    "vscode-builtin-go": "https://open-vsx.org/api/vscode/go/1.45.1/file/vscode.go-1.45.1.vsix",
    "vscode-builtin-groovy": "https://open-vsx.org/api/vscode/groovy/1.45.1/file/vscode.groovy-1.45.1.vsix",
    "vscode-builtin-grunt": "https://open-vsx.org/api/vscode/grunt/1.45.1/file/vscode.grunt-1.45.1.vsix",
    "vscode-builtin-gulp": "https://open-vsx.org/api/vscode/gulp/1.45.1/file/vscode.gulp-1.45.1.vsix",
    "vscode-builtin-handlebars": "https://open-vsx.org/api/vscode/handlebars/1.45.1/file/vscode.handlebars-1.45.1.vsix",
    "vscode-builtin-hlsl": "https://open-vsx.org/api/vscode/hlsl/1.45.1/file/vscode.hlsl-1.45.1.vsix",
    "vscode-builtin-html": "https://open-vsx.org/api/vscode/html/1.48.0-next.4a1bcdafe9/file/vscode.html-1.48.0-next.4a1bcdafe9.vsix",
    "vscode-builtin-html-language-features": "https://open-vsx.org/api/vscode/html-language-features/1.48.0-next.4a1bcdafe9/file/vscode.html-language-features-1.48.0-next.4a1bcdafe9.vsix",
    "vscode-builtin-ini": "https://open-vsx.org/api/vscode/ini/1.45.1/file/vscode.ini-1.45.1.vsix",
    "vscode-builtin-jake": "https://open-vsx.org/api/vscode/jake/1.45.1/file/vscode.jake-1.45.1.vsix",
    "vscode-builtin-java": "https://open-vsx.org/api/vscode/java/1.45.1/file/vscode.java-1.45.1.vsix",
    "vscode-builtin-javascript": "https://open-vsx.org/api/vscode/javascript/1.45.1/file/vscode.javascript-1.45.1.vsix",
    "vscode-builtin-js-debug": "https://open-vsx.org/api/ms-vscode/js-debug/1.49.8/file/ms-vscode.js-debug-1.49.8.vsix",
    "vscode-builtin-json": "https://open-vsx.org/api/vscode/json/1.46.1/file/vscode.json-1.46.1.vsix",
    "vscode-builtin-json-language-features": "https://open-vsx.org/api/vscode/json-language-features/1.46.1/file/vscode.json-language-features-1.46.1.vsix",
    "vscode-builtin-less": "https://open-vsx.org/api/vscode/less/1.45.1/file/vscode.less-1.45.1.vsix",
    "vscode-builtin-log": "https://open-vsx.org/api/vscode/log/1.45.1/file/vscode.log-1.45.1.vsix",
    "vscode-builtin-lua": "https://open-vsx.org/api/vscode/lua/1.45.1/file/vscode.lua-1.45.1.vsix",
    "vscode-builtin-make": "https://open-vsx.org/api/vscode/make/1.45.1/file/vscode.make-1.45.1.vsix",
    "vscode-builtin-markdown": "https://open-vsx.org/api/vscode/markdown/1.45.1/file/vscode.markdown-1.45.1.vsix",
    "vscode-builtin-markdown-language-features": "https://open-vsx.org/api/vscode/markdown-language-features/1.39.2/file/vscode.markdown-language-features-1.39.2.vsix",
    "vscode-builtin-merge-conflict": "https://open-vsx.org/api/vscode/merge-conflict/1.45.1/file/vscode.merge-conflict-1.45.1.vsix",
    "vscode-builtin-npm": "https://open-vsx.org/api/vscode/npm/1.45.1/file/vscode.npm-1.45.1.vsix",
    "vscode-builtin-node-debug": "https://open-vsx.org/api/ms-vscode/node-debug/1.44.8/file/ms-vscode.node-debug-1.44.8.vsix",
    "vscode-builtin-node-debug2": "https://open-vsx.org/api/ms-vscode/node-debug2/1.42.1/file/ms-vscode.node-debug2-1.42.1.vsix",
    "vscode-builtin-objective-c": "https://open-vsx.org/api/vscode/objective-c/1.45.1/file/vscode.objective-c-1.45.1.vsix",
    "vscode-builtin-perl": "https://open-vsx.org/api/vscode/perl/1.45.1/file/vscode.perl-1.45.1.vsix",
    "vscode-builtin-powershell": "https://open-vsx.org/api/vscode/powershell/1.45.1/file/vscode.powershell-1.45.1.vsix",
    "vscode-builtin-pug": "https://open-vsx.org/api/vscode/pug/1.45.1/file/vscode.pug-1.45.1.vsix",
    "vscode-builtin-python": "https://open-vsx.org/api/vscode/python/1.45.1/file/vscode.python-1.45.1.vsix",
    "vscode-builtin-r": "https://open-vsx.org/api/vscode/r/1.45.1/file/vscode.r-1.45.1.vsix",
    "vscode-builtin-razor": "https://open-vsx.org/api/vscode/razor/1.45.1/file/vscode.razor-1.45.1.vsix",
    "vscode-builtin-ruby": "https://open-vsx.org/api/vscode/ruby/1.45.1/file/vscode.ruby-1.45.1.vsix",
    "vscode-builtin-rust": "https://open-vsx.org/api/vscode/rust/1.45.1/file/vscode.rust-1.45.1.vsix",
    "vscode-builtin-scss": "https://open-vsx.org/api/vscode/scss/1.45.1/file/vscode.scss-1.45.1.vsix",
    "vscode-builtin-shaderlab": "https://open-vsx.org/api/vscode/shaderlab/1.45.1/file/vscode.shaderlab-1.45.1.vsix",
    "vscode-builtin-shellscript": "https://open-vsx.org/api/vscode/shellscript/1.45.1/file/vscode.shellscript-1.45.1.vsix",
    "vscode-builtin-sql": "https://open-vsx.org/api/vscode/sql/1.45.1/file/vscode.sql-1.45.1.vsix",
    "vscode-builtin-swift": "https://open-vsx.org/api/vscode/swift/1.45.1/file/vscode.swift-1.45.1.vsix",
    "vscode-builtin-theme-abyss": "https://open-vsx.org/api/vscode/theme-abyss/1.45.1/file/vscode.theme-abyss-1.45.1.vsix",
    "vscode-builtin-theme-defaults": "https://open-vsx.org/api/vscode/theme-defaults/1.45.1/file/vscode.theme-defaults-1.45.1.vsix",
    "vscode-builtin-theme-kimbie-dark": "https://open-vsx.org/api/vscode/theme-kimbie-dark/1.45.1/file/vscode.theme-kimbie-dark-1.45.1.vsix",
    "vscode-builtin-theme-monokai": "https://open-vsx.org/api/vscode/theme-monokai/1.45.1/file/vscode.theme-monokai-1.45.1.vsix",
    "vscode-builtin-theme-monokai-dimmed": "https://open-vsx.org/api/vscode/theme-monokai-dimmed/1.45.1/file/vscode.theme-monokai-dimmed-1.45.1.vsix",
    "vscode-builtin-theme-quietlight": "https://open-vsx.org/api/vscode/theme-quietlight/1.45.1/file/vscode.theme-quietlight-1.45.1.vsix",
    "vscode-builtin-theme-red": "https://open-vsx.org/api/vscode/theme-red/1.45.1/file/vscode.theme-red-1.45.1.vsix",
    "vscode-builtin-theme-solarized-dark": "https://open-vsx.org/api/vscode/theme-solarized-dark/1.45.1/file/vscode.theme-solarized-dark-1.45.1.vsix",
    "vscode-builtin-theme-tomorrow-night-blue": "https://open-vsx.org/api/vscode/theme-tomorrow-night-blue/1.45.1/file/vscode.theme-tomorrow-night-blue-1.45.1.vsix",
    "vscode-builtin-typescript": "https://open-vsx.org/api/vscode/typescript/1.45.1/file/vscode.typescript-1.45.1.vsix",
    "vscode-builtin-typescript-language-features": "https://open-vsx.org/api/vscode/typescript-language-features/1.45.1/file/vscode.typescript-language-features-1.45.1.vsix",
    "vscode-builtin-vb": "https://open-vsx.org/api/vscode/vb/1.45.1/file/vscode.vb-1.45.1.vsix",
    "vscode-builtin-icon-theme-seti": "https://open-vsx.org/api/vscode/vscode-theme-seti/1.45.1/file/vscode.vscode-theme-seti-1.45.1.vsix",
    "vscode-builtin-xml": "https://open-vsx.org/api/vscode/xml/1.45.1/file/vscode.xml-1.45.1.vsix",
    "vscode-builtin-yaml": "https://open-vsx.org/api/vscode/yaml/1.45.1/file/vscode.yaml-1.45.1.vsix",
    "vscode-editorconfig": "https://open-vsx.org/api/EditorConfig/EditorConfig/0.14.4/file/EditorConfig.EditorConfig-0.14.4.vsix",
    "vscode-eslint": "https://open-vsx.org/api/dbaeumer/vscode-eslint/2.1.1/file/dbaeumer.vscode-eslint-2.1.1.vsix",
    "vscode-references-view": "https://open-vsx.org/api/ms-vscode/references-view/0.0.47/file/ms-vscode.references-view-0.0.47.vsix"
  }
}

my top level directory structure is as follows : 

drwxrwxr-x   14 bobs bobs   4096 Jun 12 08:52 .
drwxrwxr-x    4 bobs bobs   4096 May 19 05:58 ..
drwxrwxr-x    5 bobs bobs   4096 Jun  7 22:04 bindery
-rw-rw-r--    1 bobs bobs 209636 May 18 06:27 CHANGELOG.md
-rw-rw-r--    1 bobs bobs   3900 May 18 06:27 CODE_OF_CONDUCT.md
drwxrwxr-x    2 bobs bobs   4096 May 19 05:47 configs
-rw-rw-r--    1 bobs bobs   4986 May 18 06:27 CONTRIBUTING.md
-rw-rw-r--    1 bobs bobs   1898 May 18 06:27 devfile.yaml
drwxrwxr-x    8 bobs bobs   4096 May 19 06:05 dev-packages
drwxrwxr-x    3 bobs bobs   4096 May 18 06:27 doc
-rw-rw-r--    1 bobs bobs    147 May 18 06:27 .editorconfig
-rw-rw-r--    1 bobs bobs    446 May 18 06:27 .eslintrc.js
drwxrwxr-x    8 bobs bobs   4096 May 19 06:10 .git
drwxrwxr-x    4 bobs bobs   4096 May 18 06:27 .github
-rw-rw-r--    1 bobs bobs    427 May 18 06:27 .gitignore
-rw-rw-r--    1 bobs bobs    771 May 18 06:27 .gitpod.dockerfile
-rw-rw-r--    1 bobs bobs    557 May 18 06:27 .gitpod.yml
-rw-rw-r--    1 bobs bobs   5842 Jun 12 08:52 lerna-debug.log
-rw-rw-r--    1 bobs bobs    277 May 18 06:27 lerna.json
-rw-rw-r--    1 bobs bobs  35406 May 18 06:27 LICENSE
drwxrwxr-x    2 bobs bobs   4096 May 18 06:27 logo
drwxrwxr-x 1046 bobs bobs  36864 Jun  8 08:05 node_modules
-rw-rw-r--    1 bobs bobs  11541 May 18 06:27 NOTICE.md
-rw-rw-r--    1 bobs bobs    175 May 18 06:27 .npmignore
-rw-rw-r--    1 bobs bobs  11554 Jun  7 22:05 package.json
drwxrwxr-x   43 bobs bobs   4096 May 18 06:27 packages
-rw-rw-r--    1 bobs bobs    498 May 18 06:27 README.md
drwxrwxr-x    2 bobs bobs   4096 May 18 06:27 scripts
drwxrwxr-x    2 bobs bobs   4096 May 18 06:27 .theia
-rw-rw-r--    1 bobs bobs   4735 Jun  7 22:09 tsconfig.json
-rw-rw-r--    1 bobs bobs     67 May 18 06:27 tsfmt.json
drwxrwxr-x    2 bobs bobs   4096 May 18 06:27 .vscode
-rw-rw-r--    1 bobs bobs 487444 Jun  7 22:14 yarn.lock
-rw-rw-r--    1 bobs bobs     40 May 18 06:27 .yarnrc

and I have changed the 'examples' folder to 'bindery'

and that folder looks like this : 

drwxrwxr-x  5 bobs bobs 4096 Jun  7 22:04 .
drwxrwxr-x 14 bobs bobs 4096 Jun 12 08:52 ..
drwxrwxr-x  5 bobs bobs 4096 Jun  7 20:55 api-samples
drwxrwxr-x  3 bobs bobs 4096 May 18 06:27 api-tests
drwxrwxr-x  3 bobs bobs 4096 Jun  8 08:05 src

where 'src' would have been 'browser'.

The package.json file in the 'src' folder is as follows : 

{
  "private": true,
  "name": "@theia/bindery",
  "version": "1.13.0",
  "license": "EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0",
  "theia": {
    "frontend": {
      "config": {
        "applicationName": "Theia Browser Example",
        "preferences": {
          "files.enableTrash": false
        }
      }
    }
  },
  "dependencies": {
    "@theia/api-samples": "1.13.0",
    "@theia/bulk-edit": "1.13.0",
    "@theia/callhierarchy": "1.13.0",
    "@theia/console": "1.13.0",
    "@theia/core": "1.13.0",
    "@theia/debug": "1.13.0",
    "@theia/editor": "1.13.0",
    "@theia/editor-preview": "1.13.0",
    "@theia/file-search": "1.13.0",
    "@theia/filesystem": "1.13.0",
    "@theia/getting-started": "1.13.0",
    "@theia/git": "1.13.0",
    "@theia/keymaps": "1.13.0",
    "@theia/markers": "1.13.0",
    "@theia/messages": "1.13.0",
    "@theia/metrics": "1.13.0",
    "@theia/mini-browser": "1.13.0",
    "@theia/monaco": "1.13.0",
    "@theia/navigator": "1.13.0",
    "@theia/outline-view": "1.13.0",
    "@theia/output": "1.13.0",
    "@theia/plugin-dev": "1.13.0",
    "@theia/plugin-ext": "1.13.0",
    "@theia/plugin-ext-vscode": "1.13.0",
    "@theia/plugin-metrics": "1.13.0",
    "@theia/preferences": "1.13.0",
    "@theia/preview": "1.13.0",
    "@theia/process": "1.13.0",
    "@theia/property-view": "1.13.0",
    "@theia/scm": "1.13.0",
    "@theia/scm-extra": "1.13.0",
    "@theia/search-in-workspace": "1.13.0",
    "@theia/task": "1.13.0",
    "@theia/terminal": "1.13.0",
    "@theia/timeline": "1.13.0",
    "@theia/typehierarchy": "1.13.0",
    "@theia/userstorage": "1.13.0",
    "@theia/variable-resolver": "1.13.0",
    "@theia/vsx-registry": "1.13.0",
    "@theia/workspace": "1.13.0"
  },
  "scripts": {
    "prepare": "yarn run clean && yarn build",
    "clean": "theia clean",
    "build": "theia build --mode development",
    "watch": "yarn build --watch",
    "start": "theia start --plugins=local-dir:../../plugins",
    "start:debug": "yarn start --log-level=debug",
    "test": "theia test . --plugins=local-dir:../../plugins --test-spec=../api-tests/**/*.spec.js",
    "test:debug": "yarn test --test-inspect",
    "coverage": "yarn test --test-coverage && yarn coverage:report",
    "coverage:report": "nyc report --reporter=html",
    "coverage:clean": "rimraf .nyc_output && rimraf coverage"
  },
  "devDependencies": {
    "@theia/cli": "1.13.0"
  }

uname -a gives : 

Linux bobs-VirtualBox 5.4.0-73-generic #82~18.04.1-Ubuntu SMP Fri Apr 16 15:10:02 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux

I'm pretty sure I'm missing something obvious and that I'm not seeing it because I have been staring at the code for far too long and I need to drink less coffee ;-)

Many thanks in advance for any help.

**Comment by vince-fugnitto** [comment]
@chris-odell-focs thank you for the discussion, please note that forking the repository and attempting to make modifications like you've done is not advised. Instead, if you are building an application using the framework you should think of [composing an application](https://theia-ide.org/docs/composing_applications/). Customizations can be contributed as [custom extensions](https://theia-ide.org/docs/authoring_extensions/).

**Reply by chris-odell-focs** [reply]
Ahhh....I did try that but had other issues(and I know that is too vague for an answer here :-) ). I'll give it another go and see how far I get.

**Reply by chris-odell-focs** [reply]
So when I follow the instructions to compose an application I always get an 

root WARN The local plugin referenced by local-dir:/home/bob/.theia/plugins does not exist.
root WARN The local plugin referenced by local-dir:/home/bob/.theia/extensions does not exist.

error.

I want to point those locations at the folder theia is running in. I have changed the package.json : 

{
  "private": true,
  "dependencies": {
    "@theia/callhierarchy": "next",
    "@theia/file-search": "next",
    "@theia/git": "next",
    "@theia/markers": "next",
    "@theia/messages": "next",
    "@theia/mini-browser": "next",
    "@theia/navigator": "next",
    "@theia/outline-view": "next",
    "@theia/plugin-ext-vscode": "next",
    "@theia/preferences": "next",
    "@theia/preview": "next",
    "@theia/search-in-workspace": "next",
    "@theia/terminal": "next"
  },
  "devDependencies": {
    "@theia/cli": "next"
  },
  "scripts": {
    "prepare": "yarn run clean && yarn build && yarn run download:plugins",
    "clean": "theia clean",
    "build": "theia build --mode development",
    "start": "theia start --plugins=/home/bob/workspace/bindery-2/plugins",
    "download:plugins": "theia download:plugins"
  },
  "theiaPluginsDir": "/home/bob/workspace/bindery-2/plugins",
  "theiaPlugins": {
    "vscode-builtin-css": "https://open-vsx.org/api/vscode/css/1.50.0/file/vscode.css-1.50.0.vsix",
    "vscode-builtin-html": "https://open-vsx.org/api/vscode/html/1.50.0/file/vscode.html-1.50.0.vsix",
    "vscode-builtin-javascript": "https://open-vsx.org/api/vscode/javascript/1.50.0/file/vscode.javascript-1.50.0.vsix",
    "vscode-builtin-json": "https://open-vsx.org/api/vscode/json/1.50.0/file/vscode.json-1.50.0.vsix",
    "vscode-builtin-markdown": "https://open-vsx.org/api/vscode/markdown/1.50.0/file/vscode.markdown-1.50.0.vsix",
    "vscode-builtin-npm": "https://open-vsx.org/api/vscode/npm/1.50.0/file/vscode.npm-1.50.0.vsix",
    "vscode-builtin-scss": "https://open-vsx.org/api/vscode/scss/1.50.0/file/vscode.scss-1.50.0.vsix",
    "vscode-builtin-typescript": "https://open-vsx.org/api/vscode/typescript/1.50.0/file/vscode.typescript-1.50.0.vsix",
    "vscode-builtin-typescript-language-features": "https://open-vsx.org/api/vscode/typescript-language-features/1.45.1/file/vscode.typescript-language-features-1.45.1.vsix"
  }
}

But when I run $ yarn theia start in the same folder as the package.json I always get the 

root WARN The local plugin referenced by local-dir:/home/bob/.theia/plugins does not exist.
root WARN The local plugin referenced by local-dir:/home/bob/.theia/extensions does not exist.

errors

If I create those folders then the errors go away but the plugins are downloaded to the plugins folder in '/home/bob/workspace/bindery-2/plugins', but I think Theia is trying to load them from '/home/bob/.theia/plugins' which is empty, and no plugins are loaded.

**Reply by vince-fugnitto** [reply]
@chris-odell-focs I believe it is expected, plugins under `/home/bob/.theia/plugins` would be user-installed plugins from the `extensions-view`. In your case you likely did not install anything so it is a simple log. The plugins you have installed (which are bundled by the application) would be the ones you referenced in your `package.json` and placed under `"theiaPluginsDir": "/home/bob/workspace/bindery-2/plugins"`.
